libs/wingsdk/src/fs/fs.ts

```diff
@@ -0,0 +1,293 @@
+   * @default "0o777"
+   */
+  readonly mode?: string;
+}
+
+/**
+ * Custom settings for removing files and directories
+ */
+export interface RemoveOptions {
+  /**
+   * When `true`, exceptions will be ignored if `path` does not exist.
+   * @default false
+   */
+  readonly force?: boolean;
+  /**
+   * If `true`, perform a recursive directory removal. In
+   * recursive mode, operations are retried on failure.
+   * @default false
+   */
+  readonly recursive?: boolean;
+}
+
+/**
+ * The fs class is used for interacting with the file system.
+ * All file paths must be POSIX file paths (/ instead of \),
+ * and will be normalized to the target platform if running on Windows.
+ */
+export class Util {
+  /**
+   * Join all arguments together and normalize the resulting path.
+   * @param paths The array of path need to join.
+   * @returns The resulting path after joining all the paths.
+   */
+  public static join(...paths: string[]): string {
+    return normalPath(path.join(...paths));
+  }
+  /**
+   * Retrieve the name of the directory from a given file path.
+   * @param p The path to evaluate.
+   * @returns The directory name of the path.
+   */
+  public static dirname(p: string): string {
+    return normalPath(path.dirname(p));
+  }
+  /**
+   * Retrieve the final segment of a given file path.
+   * @param p The path to evaluate.
+   * @returns The last portion of a path.
+   */
+  public static basename(p: string): string {
+    return path.basename(p);
+  }
+  /**
+   * Solve the relative path from {from} to {to} based on the current working directory.
+   * At times we have two absolute paths, and we need to derive the relative path from one to the other.
+   * @returns The relative path from {from} to {to}.
+   */
+  public static relative(from: string, to: string): string {
+    return normalPath(path.relative(from, to));
+  }
+  /**
+   * The right-most parameter is considered {to}. Other parameters are considered an array of {from}.
+   *
+   * Starting from leftmost {from} parameter, resolves {to} to an absolute path.
+   *
+   * If {to} isn't already absolute, {from} arguments are prepended in right to left order,
+   * until an absolute path is found. If after using all {from} paths still no absolute path is found,
+   * the current working directory is used as well. The resulting path is normalized,
+   * and trailing slashes are removed unless the path gets resolved to the root directory.
+   *
+   * @param paths A sequence of paths or path segments.
+   * @returns The resulting path after performing the resolve operation.
+   */
+  public static resolve(...paths: string[]): string {
+    return normalPath(path.resolve(...paths));
+  }
+  /**
+   * Check if the path exists.
+   * @param p The path to evaluate.
+   * @returns `true` if the path exists, `false` otherwise.
+   */
+  public static exists(p: string): boolean {
+    const normPath = localePath(p);
+    return fs.existsSync(normPath);
```

Comment:
No need to normalize paths when calling into the actual filesystem, it's safe to assume the given path is a usable path.

```suggestion
    return fs.existsSync(p);
```

Reply:
This feedback applies to other methods here as well. We should only normalize returned paths.